core/include/detray/tools/line_stepper.hpp

```diff
@@ -25,11 +25,16 @@ class line_stepper final : public base_stepper<track_t> {
     struct state : public base_type::state {
         state(track_t &t) : base_type::state(t) {}
 
+        /// Accumulated path length
+        scalar _path_length = 0.;
```

Comment:
Just as a note,  we actually need two variables:
1. path_length = summation of step_size 
2. accumulated_path_length = summation of abs(step_size)